### PR TITLE
FR-26-add-time-as-toggle

### PR DESCRIPTION
**Pull request by melonplant**
converted % to cycle through 3 different time formats. Added ETA in turns for turn based games. Added cycle / ticks ETA as 3rd option for time formats

### CODE REVIEW

**Comment by anicolao** [comment]
I am not sure if they'll keep their word but @AStonedApe said they'd remove NPA code from SAT so that the extensions would be compatible. I'd welcome a variation of your patch for [NPA](https://github.com/anicolao/npa) if you want to contribute it there. Feel free to message me on discord if you have any questions about the new tick based approach to times in NPA.

**Comment by Tsangares** [comment]
Sorry, I have not removed it yet

File: source/intel.js [hunk 1]
```diff
@@ -574,18 +574,37 @@ function NeptunesPrideAgent() {
...
     let msplus = msToTick(tick);
     return msToEtaString(msplus, prefix);
   };
-
+  let msToCycleString = function (msplus, prefix) {
+    let p = prefix !== undefined ? prefix : "ETA";
+    let cycleLength = NeptunesPride.universe.galaxy.production_rate;
+    let tickLength = NeptunesPride.universe.galaxy.tick_rate;
+    let ticksToComplete = Math.ceil(msplus / 60000 / tickLength);
+    let ttt =
```
Comment (Tsangares) [inline source/intel.js:600]: Please try to name things with words. This type of naming is unreadable especially without a comment.

File: source/intel.js [hunk 2]
```diff
@@ -1384,20 +1404,36 @@ function NeptunesPrideAgent() {
...
+            return (
+              superFormatTime(ms, mins, secs) +
+              " - " +
+              (
+                ((ms / 3600000) * 10) /
+                NeptunesPride.universe.galaxy.tick_rate
+              ).toFixed(2) +
+              " turn(s)"
+            );
+          }
```
Comment (Tsangares) [inline source/intel.js:1425]: IF its turn based only then it will use your tick format. let me try this out now.

Comment (Tsangares) [reply source/intel.js:1425]: Wow it looks great! Thank you this is actually relevant compared to the alternative. A great update!

**Comment by Tsangares** [comment]
@melonplant There was some linting stuff that occured from the Rome Linter. Next time you make a PR please `npm install` the git repository then run `npm run start` when working with your code.

**Pull request merged**